### ISSUE TITLE
New style menu-on-titlebar renders double sometimes

### ISSUE THREAD

**Issue by l0k1verloren** [Microsoft/vscode#55499]
It's not really a problem - it seems to correct itself when any significant repaint (triggered by interacting with any widget in the app) occurs but the new integrated taskbar/menu sometimes appears twice as shown in this image:

[Screenshot_from_2018-07-31_20-27-09.png](https://postimg.cc/image/x0k5fx9oh/)

Oh, this is:

Version: 1.25.1
Commit: 1dfc5e557209371715f655691b1235b6b26a06be
Date: 2018-07-11T15:40:20.190Z
Electron: 1.7.12
Chrome: 58.0.3029.110
Node.js: 7.9.0
V8: 5.8.283.38
Architecture: x64

on linux

**Comment by sbatten** [comment]
@l0k1verloren some work this milestone would definitely change the code path that could have caused this. Would you be able to experiment with the insiders version to see if you can repro? If not, could you provide any information on interactions that are likely to cause this for you?

**Comment by l0k1verloren** [comment]
I'm not sure but I stopped running insiders because of instability. However, I am familiar enough with X window managers to be able to say that replacing usually WM drawn widgets with app-level widgets is a bit fiddly. In fact, in the FOSS community, the introduction of replacing titlebars with Gtk+ widgets was quite controversial, and likely the issue is related to this part of X protocols, which are only just starting into the elimination of WM controlled window manipulation widgets, and mainly because of Gnome. There will be a lot of quirks between Gtk+ and KDE and other WM's, I don't know about KDE because I hate C++ and I hate KDE and Qt but for sure implementing these substitute title bar widgets will not be straightforward.

Note also I have observed the behaviour a bit further since this latest release, and it's not just the canvas that is rendering in the wrong place, if you try to use the menu when it is double as in the picture the cursor's real position on the canvas is shifted down as much as the titlebar is tall. The problem corrects itself usually after you trigger a refresh on the main window canvas, the quickest way is to resize or un/maximise the window, so the error in the code probably is mainly about forcing a repaint of the whole window, or more precisely, the lack of triggering this event.

It is probably relevant that this error is occurring within Gnome and with the Mutter WM, I should add. Unfortunately WMs are still not that standardised on linux.
